R/paginate.R

```diff
@@ -13,13 +13,20 @@
 #'     that is used to describe the next page.
 #'
 #' @inheritParams req_perform
-#' @param next_request A callback function that takes a two arguments (the
-#'   original request and the response) and returns:
+#' @param next_request A callback function that returns a [request] to the next
+#'   page or `NULL` if there is no next page. It takes a three arguments:
 #'
-#'   * a new [request] to request the next page or
-#'   * `NULL` if there is no next page.
-#' @param n_pages A function that extracts the total number of pages from
-#'   the [response].
+#'   1. `req`: the original request.
+#'   2. `resp`: the response of the current request.
+#'   3. `parsed`: the result of the argument `parse_resp`.
+#' @param parse_resp A function with one argument `resp` that parses the
```

Comment:
What happens if this errors?

Reply:
At the moment everything errors and you get a not so helpful error message. Let's tackle this together with #310.